### DISCUSSION TITLE
Apply same transform to image and mask?

### DISCUSSION THREAD

**Discussion by tom-osika** [Project-MONAI/MONAI#2983] [Q&A] [answered]
Hi All,

I have a question about applying random transforms. Say I have two composite transforms, one for training images (x values), and one for training masks (y values). Each of these composite transforms contain random transformations (like flipping, zooming, etc.).  I need to make sure that any random transforms that are applied to the image are also applied to the mask. Is `set_determinism` the best way to do this? The https://github.com/Project-MONAI/tutorials/blob/master/2d_segmentation/torch/unet_training_array.py example doesn't call `set_determinism`.

Thank you!

**Comment by wyli** [comment, marked as answer]
set_determinism will set the global seed. you can also use `set_random_state` of `Compose` which is more convenient for your use case. 

The `ArrayDataset` used in the demo script calls `transform.set_random_state(seed=self._seed)` to ensure the transforms are synchronised. At each `getitem`, `self._seed` is refreshed https://github.com/Project-MONAI/MONAI/blob/e223932ee88509d63407fefbddad8606200e7e69/monai/data/dataset.py#L1080

**Reply by tom-osika** [reply]
So `ArrayDatasets` will automatically be synchronized? That's what I'm using in my case.
